### PR TITLE
Optionally create an IP pool

### PR DESCRIPTION
**Pull request by jkremser**
Issue: https://github.com/giantswarm/giantswarm/issues/27164

### CODE REVIEW

File: helm/cluster-api-ipam-provider-in-cluster/templates/pools/pool.yaml [hunk 1]
```diff
@@ -0,0 +1,21 @@
...
+{{- with .Values.inClusterIpPool.end }}
+  end: {{ . }}
+{{- end }}
+{{- with .Values.inClusterIpPool.addresses }}
+  addresses:
+  {{- range . }}
+  - {{ . }}
+  {{- end }}
+{{- end }}
+{{- end }}
```
Comment (kopiczko) [inline helm/cluster-api-ipam-provider-in-cluster/templates/pools/pool.yaml:21]: Feels like equivalent of `{{ .Values.inClusterIpPool | toYaml | nindent 2}}`

File: helm/cluster-api-ipam-provider-in-cluster/values.yaml [hunk 2]
```diff
@@ -20,5 +20,14 @@ resources:
...
+  name: wc-cp-ips
+  # use either start & end or adresses
+  start: 10.10.222.222
+  end: 10.10.222.229
+  # addresses:
+  # - 10.10.222.223/28
+
 ciliumNetworkPolicy:
-  enabled: true
+  enabled: false
```
Comment (kopiczko) [inline helm/cluster-api-ipam-provider-in-cluster/values.yaml:33]: No schema?

File: helm/cluster-api-ipam-provider-in-cluster/templates/pools/pool.yaml (hunk 1 above)
Comment (jkremser) [reply helm/cluster-api-ipam-provider-in-cluster/templates/pools/pool.yaml:21]: you are right, I can simplify this

Comment (jkremser) [reply helm/cluster-api-ipam-provider-in-cluster/templates/pools/pool.yaml]: I reworked it a little bit, but `{{ .Values.inClusterIpPool | toYaml | nindent 2}}` can't be used w/o changing the structure of values, which I don't like

File: helm/cluster-api-ipam-provider-in-cluster/values.yaml (hunk 2 above)
Comment (jkremser) [reply helm/cluster-api-ipam-provider-in-cluster/values.yaml:33]: added

**Review by erkanerol — APPROVED** [review]
LGTM

**Pull request merged**